### PR TITLE
[FEATURE][symbology] Add data-defined property to font family/style for font markers

### PR DESCRIPTION
**Pull request by nirvn**
## Description

This PR adds data-defined properties for the font markers' font family and font style properties. I'm personally in need of this as part of multilingual layouts to avoid duplicating datasets, and I'm sure others will find ways to put this to good use.

Obligatory screenshot:
image

### CODE REVIEW

File: src/core/symbology/qgsmarkersymbollayer.cpp [hunk 1]
```diff
@@ -3230,6 +3232,24 @@ void QgsFontMarkerSymbolLayer::renderPoint( QPointF point, QgsSymbolRenderContex
     p->setPen( Qt::NoPen );
   }
 
+  if ( mDataDefinedProperties.isActive( QgsSymbolLayer::PropertyFontFamily ) )
+  {
+    context.setOriginalValueVariable( mFontFamily );
+    QString fontFamily = mDataDefinedProperties.valueAsString( QgsSymbolLayer::PropertyFontFamily, context.renderContext().expressionContext(), QString(), &ok );
```
Comment (nyalldawson) [inline src/core/symbology/qgsmarkersymbollayer.cpp]:
```suggestion
    QString fontFamily = mDataDefinedProperties.valueAsString( QgsSymbolLayer::PropertyFontFamily, context.renderContext().expressionContext(), mFontFamily, &ok );
```

File: src/core/symbology/qgsmarkersymbollayer.cpp [hunk 2]
```diff
@@ -3230,6 +3232,24 @@ void QgsFontMarkerSymbolLayer::renderPoint( QPointF point, QgsSymbolRenderContex
...
+    mFont.setFamily( ok ? fontFamily : mFontFamily );
+  }
+  if ( mDataDefinedProperties.isActive( QgsSymbolLayer::PropertyFontStyle ) )
+  {
+    context.setOriginalValueVariable( mFontStyle );
+    QString fontStyle = mDataDefinedProperties.valueAsString( QgsSymbolLayer::PropertyFontStyle, context.renderContext().expressionContext(), QString(), &ok );
```
Comment (nyalldawson) [inline src/core/symbology/qgsmarkersymbollayer.cpp]:
```suggestion
    QString fontStyle = mDataDefinedProperties.valueAsString( QgsSymbolLayer::PropertyFontStyle, context.renderContext().expressionContext(), mFontStyle, &ok );
```

File: src/core/symbology/qgsmarkersymbollayer.cpp [hunk 3]
```diff
@@ -3230,6 +3232,24 @@ void QgsFontMarkerSymbolLayer::renderPoint( QPointF point, QgsSymbolRenderContex
...
+    mFont.setStyleName( QgsFontUtils::translateNamedStyle( ok ? fontStyle : mFontStyle ) );
+  }
+  if ( mDataDefinedProperties.isActive( QgsSymbolLayer::PropertyFontFamily ) || mDataDefinedProperties.isActive( QgsSymbolLayer::PropertyFontStyle ) )
+  {
+    delete mFontMetrics;
```
Comment (nyalldawson) [inline src/core/symbology/qgsmarkersymbollayer.cpp]: any chance of upgrading this to a unique_ptr

**Review by nyalldawson — COMMENTED** [review]
Looks good! Got a test coming?

**Comment by Saijin-Naib** [comment]
Looks like that might make interop with ArcGIS Annotations a bit easier.

**Pull request merged**